### ISSUE TITLE
Compatibility with coq-8.5

### ISSUE THREAD

**Issue by yforster** [gmalecha/template-coq#7]
I tried to build template-coq using coq-8.5beta2

The first problem that occured was that `grammar.cma` seems to have changed its position. It is now in `lib/coq/grammar` instead of `lib/coq/parsing`. I was able to fix that by fiddling in a line in myocamlbuild.template.

But now I get

`Error while loading "~/.opam/coq-8.5/lib/coq/grammar/grammar.cma": error while linking ~/.opam/coq-8.5/lib/coq/grammar/grammar.cma.
Reference to undefined global 'Mutex'.`

**Comment by maximedenes** [comment]
Did you try the "coq-8.5" branch?

**Comment by yforster** [comment]
I did not initially, but tried now (a bit embarassing  that I did not have that idea myself...)

I can now compile everything without errors, but `demo.v` does not compile

`coqc -q -R "theories" Template -I "src" test-suite/demo.v` results in 

```
File "./test-suite/demo.v", line 22, characters 2-12:
Error: The reference quote_term was not found in the current environment.
```

**Comment by gmalecha** [comment]
@maximedenes I just checked out the coq-8.5 branch and when I build I get the following error:
```
"coqc"  -q  -R "theories" Template -I "src"   theories/Template
File "./theories/Template.v", line 3, characters 0-36:
Error: while loading src/template_plugin.cmxs:
error loading shared library: /home/gmalecha/work/research/template-coq/src/template_plugin.cmxs: undefined symbol: camlDeclarations
```

Any idea what causes this?

**Comment by maximedenes** [comment]
@yforster should be now fixed in the coq-8.5 branch, thanks for reporting.

@gmalecha this looks like a linking error due to files that were not recompiled properly. Did you try "make clean"?

**Comment by gmalecha** [comment]
@maximedenes this was my fault.
